src/Strings.f90

```diff
@@ -241,6 +247,48 @@ SUBROUTINE clear_str(this)
 ENDSUBROUTINE clear_str
 !
 !-------------------------------------------------------------------------------
+!> @brief breaks a string into partitions at the first instance of a delimiter
+!> @param this the string being partitioned
+!> @param sep the delimter used for partitioning
+!>
```

Comment:
Both versions would benefit from a bit more description on the return values, especially for those (like me) who aren't familiar with `partition`.  Just state clearly what each element of `tokens` contains, and briefly describe the results for corner cases such as `sep=''` or if `sep` is not found in `this`.